### PR TITLE
fix(sysinfo): correct memory percentage calculation on Unix systems

### LINKED ISSUE #6726: Physical Ram Usage is showing wrong value
### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### What happened?

fastfetch showing that i used 37% but omp shows 28% ram usage ...... ?!
Image

i used

```
{
          "background": "#378504",
          "foreground": "#f2f3f8",
          "invert_powerline": true,
          "style": "powerline",
          "powerline_symbol": "\ue0b2",
          "properties": {
            "display_mode": "compact"
          },
          "template": " \ue266 {{ printf \"%.0f\" .PhysicalPercentUsed }}% ",
          "type": "sysinfo"
}
```

### Theme

[1_prime.omp.json](https://github.com/user-attachments/files/22194941/1_prime.omp.json)

### What OS are you seeing the problem on?

Linux

### Which shell are you using?

bash

### Log output

```Shell
_,met$$$$$gg.           DISTRO 󰑃  Debian GNU/Linux 13 (trixie) x86_64
     ,g$$$$$$$$$$$$$$$P.       │ ├ 󰑃  Linux 6.12.41+deb13-amd64
   ,g$$P""       """Y$$.".     │ ├󰏖 󰑃  2286 (dpkg)
  ,$$P'              `$$$.     │ └ 󰑃  bash 5.2.37
',$$P       ,ggs.     `$$b:     DE/WM 󰑃  Hyprland 0.49.0 (Wayland)
`d$$'     ,$P"'   .    $$$     │ ├󰀻 󰑃  Tokyonight-Dark [Qt], Flat-Remix-Blue-Dark [GTK3]
 $$P      d$'     ,    $$P     │ ├ 󰑃  Bibata-Modern-Ice (24px)
 $$:      $$.   -    ,d$$'     │ ├ 󰑃  FantasqueSansMNFM-Bold (14pt)
 $$;      Y$b._   _,d$P'       │ └ 󰑃  kitty 0.41.1
 Y$$.    `.`"Y$$$$P"'          󰌢 SYSTEM 󰑃  HP Pavilion Laptop 15-eg1xxx
 `$$b      "-.__               │ ├󰻠 󰑃  11th Gen Intel(R) Core(TM) i5-1155G7 (8) @ 4.50 GHz
  `Y$$b                        │ ├󰻑 󰑃  Iris Xe Graphics
   `Y$$.                       │ ├󰍹 󰑃  1920x1080 @ 60Hz
     `$$b.                     │ ├󰾆 󰑃  2.75 GiB / 7.52 GiB (37%)
       `Y$$b.                  │ ├󰓡 󰑃  0 B / 7.63 GiB (0%)
         `"Y$b._               │ ├󰅐 󰑃  1 hour, 24 mins
             `""""             │ └󰍹 󰑃  1920x1080 @ 60Hz
                                AUDIO 󰑃  Tiger Lake-LP Smart Sound Technology Audio Controller Speaker
                                                                        


╭─   netchunk  ~                                                                                                                               64ms   bash   28%   3:45 PM 
╰─➜
```

### PR DESCRIPTION
**Pull request by sanki92**
### Prerequisites

- [x] I have read and understood the [contributing guide][CONTRIBUTING.md].
- [x] The commit message follows the [conventional commits][cc] guidelines.
- [x] Tests for the changes have been added (for bug fixes / features).
- [x] Docs have been added/updated (for bug fixes / features).

## Issue
Fixes #6726

## Problem
The `sysinfo` segment was displaying incorrect physical memory usage percentage on Unix/Linux systems. Users reported discrepancies where oh-my-posh showed 28% while system tools like `fastfetch` correctly showed 37% usage.

## Root Cause
The code relied on `gopsutil`'s `UsedPercent` field, which doesn't always provide accurate results. The correct calculation for memory usage should account for available memory (including reclaimable buffers and cache) rather than just free memory.

## Solution
Changed the memory percentage calculation in `terminal_unix.go` to manually compute:
```
PhysicalPercentUsed = (Total - Available) / Total × 100
```

This matches how modern system monitoring tools calculate memory usage and provides accurate results.

## Changes Made
- Modified `src/runtime/terminal_unix.go`: Updated `Memory()` function to calculate percentage correctly
- Added `src/runtime/terminal_unix_test.go`: Comprehensive unit tests covering edge cases (0%, 25%, 50%, 75%, 100%) and the specific scenario from the issue
- Updated `src/segments/sysinfo_test.go`: Added integration test for accurate memory percentage display

## Testing
- All existing tests pass
- New unit tests validate the calculation formula
- Verified against issue #6726 scenario: 7.52 GiB total with 4.74 GiB available correctly shows 36.96% (≈37%)

## Verification
```
Total Memory:     7.52 GiB
Available Memory: 4.74 GiB
Used:            2.78 GiB
Percentage:      36.96% ✓
```

[CONTRIBUTING.md]: https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/CONTRIBUTING.md
[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

File: src/runtime/terminal_unix.go [hunk 1]
```diff
@@ -166,7 +166,10 @@ func (term *Terminal) Memory() (*Memory, error) {
 	m.PhysicalTotalMemory = memStat.Total
 	m.PhysicalAvailableMemory = memStat.Available
 	m.PhysicalFreeMemory = memStat.Free
-	m.PhysicalPercentUsed = memStat.UsedPercent
+	
+	if memStat.Total > 0 {
+		m.PhysicalPercentUsed = float64(memStat.Total-memStat.Available) / float64(memStat.Total) * 100
```
[the remark this answers is not in the document]
Comment (sanki92) [reply src/runtime/terminal_unix.go]: @copilot Good catch on the potential underflow issue. I've updated the calculation to cast to float64 before subtraction with a defensive check for negative values.

Fixed in 0f57b7a ✓

**Review by JanDeDobbeleer — APPROVED** [review]

**Comment by JanDeDobbeleer** [comment]
@all-contributors please add @sanki92 for code

This is pure magic 🪄! @holopin-bot @sanki92 wizard
